packages/cadl-ranch-specs/http/server/versions/versioned/main.tsp

```diff
@@ -0,0 +1,57 @@
+import "@typespec/http";
+import "@azure-tools/cadl-ranch-expect";
+import "@azure-tools/typespec-azure-core";
+import "@typespec/versioning";
+
+using TypeSpec.Http;
+using TypeSpec.Versioning;
+
+/**
+ * Illustrates versioned server.
+ */
+@scenarioService(
+  "/server/versions/versioned",
+  {
+    versioned: Versions,
+  }
+)
+@server(
+  "{endpoint}",
+  "Testserver endpoint",
+  {
+    /**
+     * Need to be set as 'http://localhost:3000' in client.
+     */
+    endpoint: url,
+  }
+)
+namespace Server.Versions.Versioned;
+
+/**
+ * The version of the API.
+ */
+enum Versions {
+  /**
+   * The version 2022-12-01-preview.
+   */
+  @useDependency(global.Azure.Core.Versions.v1_0_Preview_2)
+  v2022_12_01_preview: "2022-12-01-preview",
+}
+
+@scenario
+@scenarioDoc("A simple operation without api-version. Expected url: '/without-api-version', it should not contain any api-version.")
+@route("/without-api-version")
+@head
+op withoutApiVersion(): OkResponse;
+
+@scenario
+@scenarioDoc("A simple operation with query api-version, whose default value is defined as '2022-12-01-preview'. Expected url: '/with-query-api-version?api-version=2022-12-01-preview'.")
+@route("/with-query-api-version")
+@head
+op withQueryApiVersion(@query apiVersion: string): OkResponse;
```

Comment:
Why this api version param not use the enum type?

Reply:
You mean `op withQueryApiVersion(@query apiVersion: SomeEnum): OkResponse;`? I'm not testing the type of apiversion. I'm testing whether the apiversion should be in the url.